### PR TITLE
[Test] Migrate test_volume_reduce_replica.py

### PR DESCRIPTION
**Pull request by srijan-sivakumar**
Migration of test case [test](https://github.com/gluster/glusto-tests/blob/2fd283331817cbe7070c820525420698da4bddbb/tests/functional/glusterd/test_volume_reduce_replica.py)

Fixes: #542

### CODE REVIEW

**Comment by srijan-sivakumar** [comment]
Need to run this in CI to test the volume creation effects after config changes to replica_count in distributed volumes.

**Comment by srijan-sivakumar** [comment]
```
(v1) [root@f05-h06-000-1029p redant]# python3 core/redant_main.py -c ../config.yml -t tests/functional/glusterd/test_volume_reduce_replica.py -ll D -sp
    ____  __________  ___    _   ________
   / __ \/ ____/ __ \/   |  / | / /_  __/
  / /_/ / __/ / / / / /| | /  |/ / / /   
 / _, _/ /___/ /_/ / ___ |/ /|  / / /    
/_/ |_/_____/_____/_/  |_/_/ |_/ /_/     
                                         

✔ Param Handling Success.
✔ Test List built
✔ Log dir creation successful.
✔ Environment setup successful.
✔ Running test case : test_volume_reduce_replica-dist-rep
✔ test_volume_reduce_replica-dist-rep Succeeded
Summary:
+---------------+-------+--------------+
|    Category   | Cases | Pass Percent |
+---------------+-------+--------------+
| nonDisruptive |   0   |      0       |
|   Disruptive  |   1   |    100.0     |
|     Total     |   1   |    100.0     |
+---------------+-------+--------------+
Table:
 test_volume_reduce_replica
 
+-------------+-------------+-------------------+
| Volume Type | Test Result |  Time taken (sec) |
+-------------+-------------+-------------------+
|   dist-rep  |     PASS    | 24.88694715499878 |
+-------------+-------------+-------------------+



Framework runtime : 30.617994785308838

✔ Tearing down successful.
```

**Comment by srijan-sivakumar** [comment]
> Need to run this in CI to test the volume creation effects after config changes to replica_count in distributed volumes.

Faced no issues. We are good to go ahead with reviews and further changes based on review comments.

File: tests/functional/glusterd/test_volume_reduce_replica.py [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+class TestCase(DParentTest):
+
+    def run_test(self, redant):
+        """
+        Test case:
+        1) Create a 2x3 replica volume.
+        2) Remove bricks in the volume to make it a 2x2 replica volume.
+        3) Remove bricks in the volume to make it a distribute volume.
+        """
+        self.brick_list = redant.get_all_bricks(self.vol_name,
```
Comment (aujjwal-redhat) [inline tests/functional/glusterd/test_volume_reduce_replica.py:37]: please add a check if bricks list is none : )

File: tests/functional/glusterd/test_volume_reduce_replica.py [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+                                                self.server_list[0])
+
+        remove_brick_list = [self.brick_list[0], self.brick_list[3]]
+        redant.remove_brick(self.server_list[0], self.vol_name,
+                            remove_brick_list, 'force', 2)
+        redant.es.set_vol_type_param(self.vol_name, 'replica_count', -1)
+
+        self.brick_list = redant.get_all_bricks(self.vol_name,
```
Comment (aujjwal-redhat) [inline tests/functional/glusterd/test_volume_reduce_replica.py:45]: same as above

**Review by aujjwal-redhat — CHANGES_REQUESTED** [review]

File: tests/functional/glusterd/test_volume_reduce_replica.py (hunk 1 above)
Comment (srijan-sivakumar) [reply tests/functional/glusterd/test_volume_reduce_replica.py:37]: Ack..

File: tests/functional/glusterd/test_volume_reduce_replica.py (hunk 2 above)
Comment (srijan-sivakumar) [reply tests/functional/glusterd/test_volume_reduce_replica.py:45]: Ack

**Comment by srijan-sivakumar** [comment]
```
(v1) [root@f05-h06-000-1029p redant]# python3 core/redant_main.py -c ../config.yml -t tests/functional/glusterd/test_volume_reduce_replica.py -ll D -sp
    ____  __________  ___    _   ________
   / __ \/ ____/ __ \/   |  / | / /_  __/
  / /_/ / __/ / / / / /| | /  |/ / / /   
 / _, _/ /___/ /_/ / ___ |/ /|  / / /    
/_/ |_/_____/_____/_/  |_/_/ |_/ /_/     
                                         

✔ Param Handling Success.
✔ Test List built
✔ Log dir creation successful.
✔ Environment setup successful.
✔ Running test case : test_volume_reduce_replica-dist-rep
✔ test_volume_reduce_replica-dist-rep Succeeded
Summary:
+---------------+-------+--------------+
|    Category   | Cases | Pass Percent |
+---------------+-------+--------------+
| nonDisruptive |   0   |      0       |
|   Disruptive  |   1   |    100.0     |
|     Total     |   1   |    100.0     |
+---------------+-------+--------------+
Table:
 test_volume_reduce_replica
 
+-------------+-------------+-------------------+
| Volume Type | Test Result |  Time taken (sec) |
+-------------+-------------+-------------------+
|   dist-rep  |     PASS    | 24.92900323867798 |
+-------------+-------------+-------------------+



Framework runtime : 31.54111075401306

✔ Tearing down successful.
```

**Review by aujjwal-redhat — APPROVED** [review]
LGTM

**Pull request merged**